utils/virtual_machines.py

```diff
@@ -52,11 +59,19 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
     data = provider_crud.get_yaml_data()
 
     deploy_args.update(vm_name=vm_name)
+    if template_name is None:
+        deploy_args.update(template=data['small_template'])
```

Comment:
Maybe you could wrap this in `except KeyError` to handle if the provider does not have small_template specified to raise eg. `ValueError` saying that the provider xyz does not have small_template key in yaml.